Sources/LanguageServerProtocol/SupportTypes/ServerCapabilities.swift

```diff
@@ -813,6 +903,77 @@ public struct SemanticTokensOptions: WorkDoneProgressOptions, Codable, Hashable
     self.full = full
     self.workDoneProgress = workDoneProgress
   }
+
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    guard case .dictionary(let dict) = dictionary["legend"],
+          let legend = SemanticTokensLegend(fromLSPDictionary: dict)
+    else {
+      return nil
+    }
+
+    self.legend = legend
+
+    self.range = nil
+
+    if case .bool(let value) = dictionary["range"] {
+      self.range = .bool(value)
+    }
+
+    if case .dictionary(let dict) = dictionary["range"],
+       let value = SemanticTokensRangeOptions(fromLSPDictionary: dict)
+    {
+      self.range = .value(value)
+    }
```

Comment:
What about changing this to a `switch` so that it’s clear that `.bool` and `.dictionary` are distinct branches?

```suggestion
    switch dictionary["range"] {
    case .bool(let value):
      self.range = .bool(value)
    case .dictionary(let dict):
      self.range = .value(SemanticTokensRangeOptions(fromLSPDictionary: dict))
    default:
      self.range = nil
    }
```

Reply:
ups. that does not compile.

Reply:
Oh, I typed it in GitHub so I forgot that `SemanticTokensRangeOptions(fromLSPDictionary:)` returned an optional.